### ISSUE TITLE
Transitions fail on dup'd copies

### ISSUE THREAD

**Issue by carlfreeland** [aasm/aasm#273]
If you dup a model, the copy won't successfully transition.  Example trivial migration task, model, and rails console lines provided.  Notice that when the copy transition is attempted, it's modifying the status of **the source id**, not the copy id.  It should update id 6, but updates id 5.  The SQL argument is cropped by the size of the available text window.

```rails generate migration CreateAasmBug status:string```
```
    class AasmBug < ActiveRecord::Base
      include AASM

      aasm :column => 'status' do
        state :begin, initial: true
        state :end

        event :finish do transitions from: [:begin], to: :end do end end
        event :restart do transitions from: [:end], to: :begin do end end
      end
    end
```

```
    Loading development environment (Rails 4.2.0)
    [1] pry(main)> source = AasmBug.create!
    => #<AasmBug:0x007ff6ff2c3c08 id: 5, status: "begin">
    [2] pry(main)> source.finish!
    [3] pry(main)> copy = source.dup
    [4] pry(main)> copy.save!
    [5] pry(main)> copy
    => #<AasmBug:0x007ff6fdeb8b78 id: 6, status: "end">
    [6] pry(main)> copy.restart!
       (0.1ms)  BEGIN
      SQL (0.3ms)  UPDATE "aasm_bugs" SET "status" = $1 WHERE "aasm_bugs"."id" = $2  [["status", "begin"], ["id", 5]]
       (1.8ms)  COMMIT
    => true
    [7] pry(main)> copy
    => #<AasmBug:0x007ff6fdeb8b78 id: 6, status: "end">
```

**Comment by reidmorrison** [comment]
If we check the current_state inside aasm:

```ruby
copy.aasm.current_state
# => :begin
```

Whereas the AR column is still the old value:

```ruby
copy.status
# => "end"
```

Appears to be an issue with the `:column` setting after the dup.

**Comment by alto** [comment]
@carlfreeland Thanks for all the details which really help a lot. I will have a look. Which version of _AASM_ are you using?

**Comment by vergenzt** [comment]
Hey @alto! We're on 4.2.0.

**Comment by alto** [comment]
@vergenzt So you've got the problem too?

**Comment by vergenzt** [comment]
Sorry, didn't specify. Yes, but Carl and I are on the same team, so it's the same instance of the problem. I just encountered it in the codebase again and realized we never responded to your question.

**Comment by lingceng** [comment]
@alto I added a PR to fix this issue. 

@carlfreeland feel free to have a test

**Comment by alto** [comment]
I just released version `4.10.0` including the changes from #325.
